qa/os/src/test/java/org/elasticsearch/packaging/test/PackagingTestCase.java

```diff
@@ -207,4 +185,87 @@ protected static Shell newShell() throws Exception {
+     */
+    public Shell.Result runElasticsearchStartCommand() throws Exception {
+        switch (distribution.packaging) {
+            case TAR:
+            case ZIP:
+                return Archives.runElasticsearchStartCommand(installation, sh);
+            case DEB:
+            case RPM:
+                return Packages.runElasticsearchStartCommand(sh);
+            case DOCKER:
+                // nothing, "installing" docker image is running it
+        }
+        return Shell.NO_OP;
+    }
+
+    public void stopElasticsearch() throws Exception {
+        distribution().packagingConditional()
+            .forPackage(() -> Packages.stopElasticsearch(sh))
+            .forArchive(() -> Archives.stopElasticsearch(installation))
+            .forDocker(/* TODO */ Platforms.NO_ACTION)
+            .run();
+    }
+
+    public void awaitElasticsearchStartup(Shell.Result result) throws Exception {
+        assertThat("Startup command should succeed", result.exitCode, equalTo(0));
+        distribution().packagingConditional()
+            .forPackage(() -> Packages.assertElasticsearchStarted(sh, installation))
+            .forArchive(() -> Archives.assertElasticsearchStarted(installation, sh))
+            .forDocker(Docker::waitForElasticsearchToStart)
+            .run();
+    }
+
+    /**
+     * Start Elasticsearch and wait until it's up and running. If you just want to run
+     * the start command, use {@link #runElasticsearchStartCommand()}.
+     * @throws Exception if Elasticsearch can't start
+     */
+    public void startElasticsearch() throws Exception {
+        awaitElasticsearchStartup(runElasticsearchStartCommand());
+    }
+
+    public void assertElasticsearchFailure(Shell.Result result, String expectedMessage) {
+
+        if (Files.exists(installation.logs.resolve("elasticsearch.log"))) {
+
+            // If log file exists, then we have bootstrapped our logging and the
+            // error should be in the logs
+            assertTrue("log file exists", Files.exists(installation.logs.resolve("elasticsearch.log")));
+            String logfile = FileUtils.slurp(installation.logs.resolve("elasticsearch.log"));
+            assertThat(logfile, containsString(expectedMessage));
+
+        } else if (distribution().isPackage() && Platforms.isSystemd()) {
+
+            // For systemd, retrieve the error from journalctl
+            assertThat(result.stderr, containsString("Job for elasticsearch.service failed"));
+            Shell.Result error = sh.run("journalctl --boot --unit elasticsearch.service");
+            assertThat(error.stdout, containsString(expectedMessage));
+
+        } else if (Platforms.WINDOWS == true) {
+
+            // In Windows, we have written our stdout and stderr to files in order to run
+            // in the background
+            String wrapperPid = result.stdout.trim();
+            sh.runIgnoreExitCode("Wait-Process -Timeout 15 -Id " + wrapperPid);
```

Comment:
Super-small nitpick - the timeout here is different from `ES_STARTUP_SLEEP_TIME_SECONDS`.

Reply:
Good catch! I think I changed that on my other branch but missed bringing it over here. Will fix.